airflow/bin/cli.py

```diff
@@ -1023,7 +1041,8 @@ def connections(args):
             session.add(new_conn)
             session.commit()
             msg = '\n\tSuccessfully added `conn_id`={conn_id} : {uri}\n'
-            msg = msg.format(conn_id=new_conn.conn_id, uri=args.conn_uri)
+            msg = msg.format(conn_id=new_conn.conn_id, uri=args.conn_uri or urlunparse((args.conn_type, '{login}:{password}@{host}:{port}'.format(
```

Comment:
IMHO, it might be less error prone if we can construct a ParseResult object and then call geturl().

Reply:
When constructing ParseResult, we need to create a netloc string from login, password, host, port, so I don't think it might be less error-prone. In geturl(), it invokes urlunparse().
https://github.com/python/cpython/blob/master/Lib/urllib/parse.py#L324

Reply:
Ack.